### PR TITLE
Add unittest for WMT example.

### PR DESCRIPTION
**Pull request by mohitreddy1996**
* Being done as part of #176 

* Refactor train.py: Introduce train_and_evaluate method with flags as arguments. This is originally proposed in #231 Further refactoring to align the example to guidelines in #231 will be done as a follow up.
* Add tensorflow-text and sentencepiece as test and examples requirements.
* input_pipeline should break out if enough chars are not accepted - This is required in the unit test as data is small.
* Remove unused flags.

### CODE REVIEW

**Review by avital — APPROVED** [review]

File: examples/wmt/train.py [hunk 1]
```diff
@@ -497,13 +485,53 @@ def tohost(x):
...
+  random_seed, jax_backend_target, model_dir, data_dir, vocab_path, vocab_size,
+  dataset_name, eval_dataset_name, batch_size, beam_size, eval_frequency,
+  num_train_steps, num_eval_steps, learning_rate, warmup_steps,
+  label_smoothing, weight_decay, max_target_length, max_eval_target_length,
+  max_predict_length, emb_dim, num_heads, num_layers, qkv_dim, mlp_dim,
+  share_embeddings, logits_via_embedding, use_bfloat16, restore_checkpoints,
+  save_checkpoints, checkpoint_freq):
+  """Executes model training and evaluation loop.
+  
+  Args:
```
Comment (andsteing) [inline examples/wmt/train.py:498]: Could we instead use the newly open-sourced `ml_collections.ConfigDict` ?

Comment (mohitreddy1996) [reply examples/wmt/train.py:498]: I think it will be better to decouple the changes. Similar thoughts by Avital on #419

Comment (andsteing) [reply examples/wmt/train.py:497]: Decoupling SGTM. I'm not a big fan of these huge signatures. If we refactored the examples first to use `ConfigDict` and then added the unit tests, we would never have to introduce them. Feel free to decide yourself in which order you prefer to do the two changes...

Comment (mohitreddy1996) [reply examples/wmt/train.py:497]: Yeah agreed. My thinking behind doing this before using ConfigDicts was that we will have a unit test in place as a sanity check for the transition from flags to configs. WDYT? 

I was leaning towards this solution unit test -> follow up PR with ConfigDicts.

**Comment by avital** [comment]
This is for the pre-Linen examples, so I'll close this PR.

Looks like the Linen WMT example has a test now: https://github.com/google/flax/blob/master/linen_examples/wmt/train_test.py

**Pull request closed without merging**